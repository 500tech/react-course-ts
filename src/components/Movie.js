import * as React from 'react';

const Movie = ({ label }) => <li>{label}</li>;

export default Movie;
