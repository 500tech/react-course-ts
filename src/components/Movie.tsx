import * as React from 'react';

interface ComponentProps {
  movie: {
    label: string,
    rating: number,
    id: number
  },
  selectMovie: (id) => void,
  removeMovie: (id) => void
}

const Movie = (props: ComponentProps) => (
  <li onClick={() => props.selectMovie(props.movie.id)}>
    <div>{props.movie.label}</div>
    <div
      className="remove"
      onClick={() => props.removeMovie(props.movie.id)}>
      remove
    </div>
  </li>
);

export default Movie;
