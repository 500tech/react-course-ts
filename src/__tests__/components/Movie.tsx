import * as React from 'react';
import { shallow } from 'enzyme';
import { noop } from 'lodash';
import Movie from "../../components/Movie";

const setup = ({ movie = null, selectMovie = noop, removeMovie = noop } = {}) => ({
  movie,
  selectMovie,
  removeMovie
});

describe('<Movie/>', () => {

  const movie = {
    label: 'test movie',
    rating: 1,
    id: 1
  };

  test('should render', () => {
    const props = setup({ movie });
    const component = shallow(
        <Movie {...props} />
    );

    expect(component).toMatchSnapshot();
  });

  test('should select movie', () => {
    const selectMovieSpy = jest.fn();
    const props = setup({ movie, selectMovie: selectMovieSpy });
    const component = shallow(
        <Movie {...props} />
    );

    expect(props.selectMovie).not.toHaveBeenCalled();

    component.simulate('click');

    expect(props.selectMovie).toHaveBeenCalled();
  });

  test('should remove movie', () => {
    const removeMovieSpy = jest.fn();
    const props = setup({ movie, removeMovie: removeMovieSpy });
    const component = shallow(
        <Movie {...props} />
    );

    expect(props.removeMovie).not.toHaveBeenCalled();

    component.find('.remove').simulate('click');

    expect(props.removeMovie).toHaveBeenCalled();
  });


});
