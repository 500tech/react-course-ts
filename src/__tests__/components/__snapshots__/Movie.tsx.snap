// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie/> should render 1`] = `
<li
  onClick={[Function]}
>
  <div>
    test movie
  </div>
  <div
    className="remove"
    onClick={[Function]}
  >
    remove
  </div>
</li>
`;
